src/sentry/integrations/types.py

```diff
@@ -71,3 +73,23 @@ class ExternalProviderEnum(Enum):
     ExternalProviderEnum.SLACK.value,
     ExternalProviderEnum.MSTEAMS.value,
 ]
+
+
+class EventLifecycleOutcome(Enum):
+    STARTED = "STARTED"
+    HALTED = "HALTED"
+    SUCCESS = "SUCCESS"
+    FAILURE = "FAILURE"
+
+    def __str__(self) -> str:
+        return self.value.lower()
+
+
+T = TypeVar("T")
+
+
+@dataclass
+class MessagingResponse(Generic[T]):
```

Comment:
Let's make this more generic. `IntegrationResponse` maybe? The concrete type with `T` as a `str` would make more sense as the `MessagingResponse` type.

Reply:
Discussed via slack. This is very messaging specific for now, we'll explore the generic version in the future.